plugin/parser/differ/mysql/differ.go

```diff
@@ -383,3 +384,293 @@ func getIndexName(index *ast.Constraint) string {
+	//   | {DATA | INDEX} DIRECTORY [=] 'absolute path to directory'
+	//   | DELAY_KEY_WRITE [=] {0 | 1}
+	//   | ENCRYPTION [=] {'Y' | 'N'}
+	//   | ENGINE [=] engine_name
+	//   | ENGINE_ATTRIBUTE [=] 'string'
+	//   | INSERT_METHOD [=] { NO | FIRST | LAST }
+	//   | KEY_BLOCK_SIZE [=] value
+	//   | MAX_ROWS [=] value
+	//   | MIN_ROWS [=] value
+	//   | PACK_KEYS [=] {0 | 1 | DEFAULT}
+	//   | PASSWORD [=] 'string'
+	//   | ROW_FORMAT [=] {DEFAULT | DYNAMIC | FIXED | COMPRESSED | REDUNDANT | COMPACT}
+	//   | START TRANSACTION	// This is an internal-use table option.
+	//							// It was introduced in MySQL 8.0.21 to permit CREATE TABLE ... SELECT to be logged as a single,
+	//							// atomic transaction in the binary log when using row-based replication with a storage engine that supports atomic DDL.
+	//   | SECONDARY_ENGINE_ATTRIBUTE [=] 'string'
+	//   | STATS_AUTO_RECALC [=] {DEFAULT | 0 | 1}
+	//   | STATS_PERSISTENT [=] {DEFAULT | 0 | 1}
+	//   | STATS_SAMPLE_PAGES [=] value
+	//   | TABLESPACE tablespace_name [STORAGE {DISK | MEMORY}]
+	//   | UNION [=] (tbl_name[,tbl_name]...)
+	// }
+
+	// We use map to record the table options, so we can easily find the difference.
+	oldOptionsMap := buildTableOptionMap(old)
+	newOptionsMap := buildTableOptionMap(new)
+
+	var options []*ast.TableOption
+	for oldTp, oldOption := range oldOptionsMap {
+		newOption, ok := newOptionsMap[oldTp]
+		if !ok {
+			// We should drop the table option if it doesn't exist in the new table options.
+			if astOption := dropTableOption(oldOption); astOption != nil {
+				options = append(options, astOption)
+			}
+			continue
+		}
+		if !isTableOptionValEqual(oldOption, newOption) {
+			options = append(options, newOption)
+		}
+	}
+	// We should add the table option if it doesn't exist in the old table options.
+	for newTp, newOption := range newOptionsMap {
+		if _, ok := oldOptionsMap[newTp]; !ok {
+			// We should add the table option if it doesn't exist in the old table options.
+			options = append(options, newOption)
+		}
+	}
+	if len(options) == 0 {
+		return nil
+	}
+	return &ast.AlterTableStmt{
+		Table: tableName,
+		Specs: []*ast.AlterTableSpec{
+			{
+				Tp:      ast.AlterTableOption,
+				Options: options,
+			},
+		},
+	}
+}
+
+// dropTableOption generate the table options node need to oppended to the ALTER TABLE OPTION spec.
+func dropTableOption(option *ast.TableOption) *ast.TableOption {
+	switch option.Tp {
+	case ast.TableOptionAutoIncrement:
+		// You cannot reset the counter to a value less than or equal to the value that is currently in use.
+		// For both InnoDB and MyISAM, if the value is less than or equal to the maximum value currently in the AUTO_INCREMENT column,
+		// the value is reset to the current maximum AUTO_INCREMENT column value plus one.
+		// https://dev.mysql.com/doc/refman/8.0/en/alter-table.html
+		// Users want to drop the AUTO_INCREMENT, they need to `ALTER TABLE tbl MODIFY COLUMN` to drop the AUTO_INCREMENT.
```

Comment:
Why do you remove the code here? I think resetting AutoIncrement is ok here.

Reply:
Actually, if we diff two statement
`CREATE TABLE tbl(id INT PRIMARY KEY AUTO_INCREMENT) AUTO_INCREMENT = 5`
`CREATE TABLE tbl(id INT PRIMARY KEY AUTO_INCREMENT)`
I'm not sure what is the reason for the user to drop this option.
But I guess reset never seems to be wrong. LGTM.